src/main/java/org/fogbowcloud/saps/engine/core/archiver/Archiver.java

```diff
@@ -202,7 +202,7 @@ private void garbageCollector() {
 	 * @throws Exception
 	 */
 	private void cleanUnfinishedArchivedData() throws Exception {
-		List<SapsImage> archivingTasks = tasksInArchivingState(Catalog.UNLIMITED);
+		List<SapsImage> archivingTasks = tasksInArchivingState(CatalogConstants.UNLIMITED);
```

Comment:
what is the point of having the UNLIMITED on the CatalogConstants interfaces? also, it seems odd that we need this limit only in one method of the Catalog interface `getTasksByState`
before modifying the code, please, explain why do we need this. if we really need this in this method, why don't we need in the other ones that return a collection of SapsImage? And, if we really need this, it seems better to have two methods, one that does not have the `limit` parameter and a second one that has (and, please delete this UNLIMITED flag)

Reply:
I analyzed the code and this constant was only useful in the old SAPS design by the InputDownloader component (which currently no longer exists) in the step of selecting SAPS tasks due to the disk space used by the downloaded or processed data, making it select up to X depending the situation.